tests/Operation/AggregateFunctionalTest.php

```diff
@@ -103,12 +103,29 @@ public function testTypeMapOption(array $typeMap = null, array $expectedDocument
         $this->createFixtures(3);
 
         $pipeline = [['$match' => ['_id' => ['$ne' => 2]]]];
+
         $operation = new Aggregate($this->getDatabaseName(), $this->getCollectionName(), $pipeline, ['typeMap' => $typeMap]);
         $results = iterator_to_array($operation->execute($this->getPrimaryServer()));
 
         $this->assertEquals($expectedDocuments, $results);
     }
 
+    public function testTypeMapInline()
+    {
+        if (version_compare($this->getServerVersion(), '3.6.0', '>=')) {
+            $this->markTestSkipped('Inline aggregations are not supported');
+        }
+
+        $this->createFixtures(3);
+
+        $pipeline = [['$match' => ['x' => 1]], ['$out' => $this->getCollectionName() . '.output']];
+
+        $operation = new Aggregate($this->getDatabaseName(), $this->getCollectionName(), $pipeline, ['typeMap' => ['root' => 'array'], 'useCursor' => false]);
+        $results = $operation->execute($this->getPrimaryServer());
+
+        $this->assertTrue($results instanceof \MongoDB\Model\TypeMapArrayIterator);
```

Comment:
`assertInstanceOf()` can be used here. Reminder that its class name string won't need the leading `\`, since it assumes the global namespace (like `@expectedException`).